.github/workflows/nightly_release.yml

```diff
@@ -0,0 +1,93 @@
+name: Nightly Builds
+on:
+  schedule:
+    # Runs every weekday at 1am
+    - cron: 0 1 * * 1-5
+  workflow_dispatch:
+
+
+jobs:
+  tag_nightly_release:
+    name: Tag a new nightly release
+    runs-on: ubuntu-latest
+    outputs:
+      tag_name: ${{ steps.set_tagname.outputs.tag }}
+
+    steps:
+      - name: Checkout git repository 🕝
+        uses: actions/checkout@v3
+
+      - name: Compose tag name
+        id: set_tagname
+        run: |
+          DATE=$(date +'%Y%m%d')
+
+          # Find latest rasa-oss version
+          echo "Trying to find the latest rasa-oss version..."
+          LATEST_RASA_VERSION=$(python -c "import sys; import os; sys.path.append('src/rasa/rasa'); from version import __version__; print(__version__)")
+          echo "Current RASA version: ${LATEST_RASA_VERSION}"
+
+          LATEST_NIGHTLY_VERSION=$(echo ${LATEST_RASA_VERSION} | awk -F. '{$(NF-1) = $(NF-1) + 1;} {$(NF) = 0;} 1' | sed 's/ /./g')
+
+          echo "New nightly release version: ${LATEST_NIGHTLY_VERSION}"
+
+          echo "Composing nightly build tag name..."
+          GH_TAG=${LATEST_NIGHTLY_VERSION}-dev${DATE}
+          echo ${GH_TAG}
+          echo "::set-output name=tag::${GH_TAG}"
+
+      - name: Tag latest main commit as nightly
+        uses: actions/github-script@v5
+        with:
+          github-token: ${{ secrets.RASABOT_GITHUB_TOKEN }}
+          script: |
+            github.rest.git.createRef({
```

Comment:
Git tags can have messages (and are then called annotated tags), so hopefully that should be possible here.

Reply:
Turns out with the `github.rest.git.createRef` script i cannot add a tag. I can move this tagging and pushing into the nightly script if we need to annotate it. @losterloh let me know if I should do that.

Reply:
sounds good to me @sanchariGr! Generally, I'd say something like "the more complex the logic, the more we should see about putting it in scripts instead of directly in the github actions workflows".